packages/commerce-sdk-react/src/hooks/ShopperBaskets/cache.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+    ShopperBaskets,
+    ShopperBasketsTypes,
+    ShopperCustomers,
+    ShopperCustomersTypes
+} from 'commerce-sdk-isomorphic'
+import {ApiClients, Argument, CacheUpdate, CacheUpdateMatrix, MergedOptions} from '../types'
+import {and, matchesPath, matchParameters, pick} from '../utils'
+
+type Client = ApiClients['shopperBaskets']
+/** Data returned by every Shopper Baskets endpoint (except `deleteBasket`) */
+type Basket = ShopperBasketsTypes.Basket
+/** Data returned by `getCustomerBaskets` */
+type CustomerBasketsResult = ShopperCustomersTypes.BasketsResult
+/** Parameters that get passed around, includes client config and possible parameters from other endpoints */
+type BasketParameters = MergedOptions<Client, Argument<Client['getBasket']>>['parameters']
+/** Parameters that we actually send to the API for `getBasket` */
+type GetBasketParameters = Argument<ShopperBaskets<{shortCode: string}>['getBasket']>['parameters']
+/** Parameters that we actually send to the API for `getCustomerBaskets` */
+type GetCustomerBasketsParameters = Argument<
+    ShopperCustomers<{shortCode: string}>['getCustomerBaskets']
+>['parameters']
+
+// Path helpers to avoid typos!
+const getBasePath = (parameters: {organizationId: string}) => [
+    '/organizations/',
+    parameters.organizationId
+]
+const getBasketPath = (parameters: GetBasketParameters) => [
+    ...getBasePath(parameters),
+    '/baskets/',
+    parameters.basketId
+]
+const getCustomerBasketsPath = (parameters: GetCustomerBasketsParameters) => [
+    ...getBasePath(parameters),
+    '/customers/',
+    parameters.customerId,
+    '/baskets' // No trailing / as it's an aggregate endpoint
+]
```

Comment:
If we have the leading slash for every path segment, then we never need to worry about trailing slashes. 

```suggestion
const getCustomerBasketsPath = (parameters: GetCustomerBasketsParameters) => [
    ...getBasePath(parameters),
    '/customers',
    `/${parameters.customerId}`,
    '/baskets' // No trailing / as it's an aggregate endpoint
]
```

Reply:
The trade off there is that then we don't have array elements that are _just_ the parameters. Which isn't much of a trade off, so 🤷. In any case, the introduction of the query key helpers completely eliminates the need to ever even know what the query key actually is, so I'll mark this as resolved. With the helpers, changing the query key format becomes pretty trivial, so we can always revisit.